test/tests/functional/pbs_cray_suspend_resume.py

```diff
@@ -93,3 +97,390 @@ def test_exclusive_job_not_suspended(self):
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j1id = self.server.submit(j1)
+
+        # Ensure j2 is started after j1 due to preempt_sort:
+        # min_time_since_start
+        time.sleep(1)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j1id)
+        self.assertTrue(d)
+
+        j2 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=6',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j2id = self.server.submit(j2)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j2id)
+        self.assertTrue(d)
+
+        j3 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=6',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'B',
+                             ATTR_queue: 'expressq'})
+        j3id = self.server.submit(j3)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j3id)
+        self.assertTrue(d)
+        d = self.server.expect(JOB, {ATTR_state: 'S'}, id=j2id)
+        self.assertTrue(d)
+
+        j4 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=6',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'B'})
+        j4id = self.server.submit(j4)
+        d = self.server.expect(JOB, {ATTR_state: 'Q'}, id=j4id)
+
+    def test_preempted_topjob_calendared(self):
+        """
+        That even if topjob_ineligible is set for
+        a preempted job and sched_preempt_enforce_resumption
+        is set true , the preempted job will be calendared
+        """
+
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_preempt_enforce_resumption': 'true'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '2'})
+
+        nodes = self.server.status(NODE)
+        for n in nodes:
+            if 'resources_available.vntype' in n:
+                if n['resources_available.vntype'] == 'cray_compute':
+                    ncpus = n['resources_available.ncpus']
+
+        # Submit a job
+        j = Job(TEST_USER, {ATTR_l + '.walltime': '120',
+                            ATTR_l + '.ncpus': ncpus})
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid1)
+
+        # Alter topjob_ineligible for runnng job
+        self.server.alterjob(jid1, {ATTR_W: "topjob_ineligible = true"},
+                             runas=ROOT_USER, logerr=True)
+
+        # Create a high priority queue
+        a = {'queue_type': 'e', 'started': 't',
+             'enabled': 't', 'priority': '150'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id="highp")
+
+        # Submit 2 jobs to high priority queue
+        j = Job(TEST_USER, {ATTR_queue: 'highp', ATTR_l + '.walltime': '60'})
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
```

Comment:
You need a new job object for jid3.

Reply:
The same job object can be submitted one or more times.